### PR TITLE
Render tokens pages in main nav using standard markdown renderer

### PR DESCRIPTION
**Pull request by jesstelford**
In pursuit of [speeding up CI](https://github.com/Shopify/polaris-internal/issues/1217); I am attempting to tackle the slow OG Image generation. The OG Image generation depends on [reading the generated `sitemap.xml`](https://github.com/Shopify/polaris/blob/7e6562d713e10ebf81337f8185644961f00803b8/polaris.shopify.com/scripts/gen-og-images.mjs#L176-L179) which is created by [loading up the site in dev mode and scraping it for all URLs it can see](https://github.com/Shopify/polaris/blob/main/polaris.shopify.com/scripts/gen-site-map.mjs) (_slooooowwww_).

To speed up the `sitemap.xml` generation, I want to leverage the `.cache/site.ts` data we're already generating (_fast!_), but that's based on scanning for `.mdx` files, and not everything is a `.mdx` file (yet).

Here's the URLs present in `sitemap.xml` but missing in `.cache/site.ts`:

```diff
- "/patterns-legacy",
- "/patterns/date-picking/date-list",
- "/patterns/date-picking/date-range",
- "/patterns/date-picking/single-date",
- "/sandbox",
- "/tokens/border",
- "/tokens/breakpoints",
- "/tokens/color",
- "/tokens/font",
- "/tokens/height",
- "/tokens/motion",
- "/tokens/shadow",
- "/tokens/space",
- "/tokens/text",
- "/tokens/width",
- "/tokens/z-index",
```

The first step in that is getting the `/tokens/*` pages to be `.mdx` files. Hence this PR.

One added benefit: The individual token groups now show up in the main nav! 🎉 

Screenshot 2023-12-08 at 20 25 57

Check it out: https://polaris-staging.shopifycloud.com/tokens/breakpoints

### CODE REVIEW

File: polaris.shopify.com/src/components/TokensNav/TokensNav.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    | 'breakpoints'
+    | 'color'
+    | 'font'
+    | 'height'
+    | 'motion'
+    | 'shadow'
+    | 'space'
+    | 'text'
+    | 'width'
+    | 'zIndex';
```
Comment (aaronccasanova) [inline polaris.shopify.com/src/components/TokensNav/TokensNav.tsx:17]: Understanding this is copy/paste, can we derive this from the `Theme` type?

```ts
import type {Theme} from '@shopify/polaris-tokens';

interface TokenNavProps {
  selected?: keyof Theme;
}
```

File: polaris.shopify.com/pages/[...slug].tsx [hunk 2]
```diff
@@ -310,6 +312,25 @@ export const getStaticProps: GetStaticProps<Props, {slug: string[]}> = async ({
...
+            .map(([name, value]) => ({name, ...value}))
+            // Some tokens have custom sorting for display
+            .sort((token) =>
+              token.name.includes('ease') || token.name.includes('linear')
+                ? -1
+                : 1,
+            ),
+        );
+        scope.tokens = tokenGroupObjects;
+        end();
```
Comment (aaronccasanova) [inline polaris.shopify.com/pages/[...slug].tsx:331]: Light suggestion: Can we avoid changing the shape of the `metaTheme` object? I'd personally prefer folks to use and transform the `metaTheme` directly for consistency e.g.

```mdx
---
title: Text
showTOC: false
---

# Tokens &rarr; {frontmatter.title}

<TokensNav selected="text" />

<TokenList>
  {Object.entries(metaTheme.text)
    .map(([tokenName, tokenProperties]) => (
      <TokenList.Item
        key={tokenName}
        category="text"
        name={tokenName}
        value={tokenProperties.value}
        description={tokenProperties.description}
      />
    ))}
</TokenList>
```

**Review by aaronccasanova — APPROVED** [review]
Updates looks awesome @jesstelford 🚀

File: polaris.shopify.com/src/components/TokensNav/TokensNav.tsx (hunk 1 above)
Comment (jesstelford) [reply polaris.shopify.com/src/components/TokensNav/TokensNav.tsx:17]: Great catch! Fixed that up [here](https://github.com/Shopify/polaris/pull/11306/files#diff-61dbaaee52e7841d2cf7a4f18606bc42759fe6fd514d29e7eeac60b94dfcd03cR1)

File: polaris.shopify.com/pages/[...slug].tsx (hunk 2 above)
Comment (jesstelford) [reply polaris.shopify.com/pages/[...slug].tsx:331]: A good idea, and one I toyed with a bit at first, but I eventually decided against it due to the added JS-ification of the MDX... I'm worried it's getting too far from "Markdown with some React sprinkled in".

For now, I'd like to leave it as-is, but am only 50/50 opinionated on this one.

**Pull request merged**